File: 0190-reverse-bits/0190-reverse-bits.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        int  rev = 0;
        int digit;
        
        for(int i=0;i<32;i++){
            int bit = n&1;
            int bitSift = bit <<(31-i);
            rev = rev | bitSift;
             n = n >> 1;
        }
        return rev;
    }
}

File: 0190-reverse-bits/0190-reverse-bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        
       uint32_t  rev = 0;
        int digit;
        
        for(int i=0;i<32;i++){
            uint32_t bit = n&1;
            uint32_t bitSift = bit <<(31-i);
            rev = rev | bitSift;
            n= n>>1;
        }
        return rev;
    }
};